### ISSUE TITLE
Some tests fail on GPU

### ISSUE THREAD

**Issue by elcorto** [lightning-uq-box/lightning-uq-box#145]
I see some tests fail when running on a GPU box. Logs are attached. Summary:

```sh
$ grep -E '^E[ ]+RuntimeError' logs_gpu_failed | sort -u
E       RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu! (when checking argument for argument mat1 in method wrapper_CUDA_addmm)
E       RuntimeError: Input type (torch.FloatTensor) and weight type (torch.cuda.FloatTensor) should be the same or input should be a MKLDNN tensor and weight is a dense tensor
```

This is with torch 2.2.0+rocm5.7 on an AMD GPU machine. Haven't tested on Nvidia, though.

[logs_gpu_failed.txt](https://github.com/lightning-uq-box/lightning-uq-box/files/15422315/logs_gpu_failed.txt)

**Comment by nilsleh** [comment]
Mhm that's a bit strange since all the tests are configured to run on cpu by default with `accelerator=cpu`. We did this because tests are failing due to memory issues in CI otherwise #139 . So I don't immediately see how the tests can be run on GPU, without overwriting the tests.

**Comment by elcorto** [comment]
Yes, that's strange then. Maybe not all tests use `accelerator=cpu`? I had to switch the GPU off by `CUDA_VISIBLE_DEVICES= pytest ...` in order for it to not get picked up.
